README.md
A Flutter package that creates a clean and minimalist dialog Supports null-safety and Flutter 3

## Features

- Customizable
- Lightweight 

## Getting started 

- Install the package through this command 

```bash 
flutter pub get clean_dialog
```

- Import the package inside the file you desire to include the dialog.

```dart 
import 'package:clean_dialog/clean_dialog.dart';
```

- Add this line to the onPress function of the desired Button or gesture detector.

```dart
showDialog(
    context: context,
    builder: (context) => CleanDialog(
    title: 'Error',
    content: 'We were not able to update your information.',
    backgroundColor: const Color(0XFFbe3a2c),
    titleTextStyle: const TextStyle(fontSize: 25, fontWeight: FontWeight.bold, color: Colors.white),
    contentTextStyle: const TextStyle(fontSize: 16, color: Colors.white),
    actions: [
        CleanDialogActionButtons(
            actionTitle: 'Cancel',
            onPressed: () => Navigator.pop(context),
        ),
        CleanDialogActionButtons(
            actionTitle: 'Try again',
            textColor: const Color(0XFF27ae61),
            onPressed: () {},
        ),
    ],
    ),
);
```
##  Output

<img src="https://github.com/OmarMach/clean_dialog/raw/main/doc/exp.png" width="400" alt="Example 1">
<img src="https://github.com/OmarMach/clean_dialog/raw/main/doc/exp2.png" width="400" alt="Example 2">





CHANGELOG.md
## 0.0.1

* Created package.
## 0.0.12

* Update package.

## 0.0.13

* Fix readme.

## 0.0.14

* Fix readme2.

## 0.0.15

* Fix format/License.

## 0.0.16

* Added documentation.

test/clean_dialog_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:clean_dialog/clean_dialog.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
  });
}


lib/clean_dialog.dart
library clean_dialog;

import 'package:flutter/material.dart';

/// A Clean and minimalist Flutter Dialog
/***
 * CleanDialog(
    title: 'Error',
    content: 'We were not able to update your information.',
    backgroundColor: const Color(0XFFbe3a2c),
    titleTextStyle: const TextStyle(fontSize: 25, fontWeight: FontWeight.bold, color: Colors.white),
    contentTextStyle: const TextStyle(fontSize: 16, color: Colors.white),
    actions: [
        CleanDialogActionButtons(
            actionTitle: 'Cancel',
            onPressed: () => Navigator.pop(context),
        ),
        CleanDialogActionButtons(
            actionTitle: 'Try again',
            textColor: const Color(0XFF27ae61),
            onPressed: () {},
        ),
    ],
    ),
 */
///
class CleanDialog extends StatelessWidget {
  const CleanDialog({
    super.key,
    required this.backgroundColor,
    required this.actions,
    required this.title,
    required this.content,
    this.titleTextStyle,
    this.contentTextStyle,
    this.titleTextAlign,
    this.contentTextAlign,
  });

  /// Color to be displayed in the background of the Dialog
  final Color backgroundColor;

  /// The String to be displayed as title for the Dialog
  final String title;

  /// The dialog's text content
  final String content;

  /// The list of Buttons that will be displayed in the Dialog
  final List<CleanDialogActionButtons> actions;

  /// The title text alignement default value is TextAlign.Center
  final TextAlign? titleTextAlign;

  /// The content text alignement default value is TextAlign.Center
  final TextAlign? contentTextAlign;

  /// The content text style default value is TextStyle(fontWeight: FontWeight.bold, fontSize: 23)
  final TextStyle? titleTextStyle;

  /// The content text style default value is TextStyle(fontSize: 16)
  final TextStyle? contentTextStyle;

  /// Used to transfer backgroundColor to buttons textColor.
  List<CleanDialogActionButtons> _buildStyledActionButtons() => actions
      .map((e) => e.copyWith(textColor: e.textColor ?? backgroundColor))
      .toList();

  @override
  Widget build(BuildContext context) {
    final List<CleanDialogActionButtons> actions = _buildStyledActionButtons();

    return Dialog(
      elevation: 0,
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(20)),
      child: ClipRRect(
        borderRadius: BorderRadius.circular(20),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            Container(
              color: backgroundColor,
              child: Padding(
                padding: const EdgeInsets.all(15.0),
                child: Column(
                  children: [
                    Padding(
                      padding: const EdgeInsets.all(8.0),
                      child: Text(
                        title,
                        textAlign: titleTextAlign ?? TextAlign.center,
                        style: titleTextStyle ??
                            const TextStyle(
                                fontWeight: FontWeight.bold, fontSize: 23),
                      ),
                    ),
                    Text(
                      content,
                      textAlign: contentTextAlign ?? TextAlign.center,
                      style: contentTextStyle ?? const TextStyle(fontSize: 16),
                    ),
                  ],
                ),
              ),
            ),
            ListView.separated(
              physics: const NeverScrollableScrollPhysics(),
              shrinkWrap: true,
              itemBuilder: (context, index) => actions.elementAt(index),
              separatorBuilder: (context, _) => Divider(
                color: backgroundColor.withOpacity(.5),
                height: 1,
              ),
              itemCount: actions.length,
            )
          ],
        ),
      ),
    );
  }
}

class CleanDialogActionButtons extends StatelessWidget {
  const CleanDialogActionButtons(
      {super.key,
      required this.actionTitle,
      required this.onPressed,
      this.textColor});

  /// The String text that will be displayed for the action.
  final String actionTitle;

  /// The action that will be executed once the button tapped.
  final VoidCallback onPressed;

  /// Optional Color that will override the default parent `backgoundColor`.
  final Color? textColor;

  CleanDialogActionButtons copyWith({
    String? actionTitle,
    VoidCallback? onPressed,
    Color? textColor,
  }) =>
      CleanDialogActionButtons(
        actionTitle: actionTitle ?? this.actionTitle,
        onPressed: onPressed ?? this.onPressed,
        textColor: textColor ?? this.textColor,
      );

  @override
  Widget build(BuildContext context) {
    return InkWell(
      splashColor: textColor?.withOpacity(.1),
      highlightColor: textColor?.withOpacity(.2),
      onTap: onPressed,
      child: Padding(
        padding: const EdgeInsets.all(15.0),
        child: Text(
          actionTitle,
          textAlign: TextAlign.center,
          style: TextStyle(
              color: textColor ?? Colors.blue, fontWeight: FontWeight.bold),
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:clean_dialog/clean_dialog.dart';
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Clean Dialog example app',
      home: MainScreen(),
    );
  }
}

class MainScreen extends StatelessWidget {
  const MainScreen({
    super.key,
  });

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Padding(
        padding: const EdgeInsets.all(8.0),
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.center,
          children: [
            ElevatedButton(
              onPressed: () {
                showDialog(
                  context: context,
                  builder: (context) => CleanDialog(
                    title: 'Success',
                    content: 'You have successfully updated your information.',
                    backgroundColor: const Color(0XFF27ae61),
                    titleTextStyle: const TextStyle(
                        fontSize: 25,
                        fontWeight: FontWeight.bold,
                        color: Colors.white),
                    contentTextStyle:
                        const TextStyle(fontSize: 16, color: Colors.white),
                    actions: [
                      CleanDialogActionButtons(
                          actionTitle: 'Accept', onPressed: () {}),
                      CleanDialogActionButtons(
                        actionTitle: 'Cancel',
                        onPressed: () => Navigator.pop(context),
                        textColor: const Color(0XFFbe3a2c),
                      ),
                    ],
                  ),
                );
              },
              child: const Center(child: Text('First Example')),
            ),
            ElevatedButton(
              onPressed: () {
                showDialog(
                  context: context,
                  builder: (context) => CleanDialog(
                    title: 'Error',
                    content: 'We were not able to update your information.',
                    backgroundColor: const Color(0XFFbe3a2c),
                    titleTextStyle: const TextStyle(
                        fontSize: 25,
                        fontWeight: FontWeight.bold,
                        color: Colors.white),
                    contentTextStyle:
                        const TextStyle(fontSize: 16, color: Colors.white),
                    actions: [
                      CleanDialogActionButtons(
                        actionTitle: 'Cancel',
                        onPressed: () => Navigator.pop(context),
                      ),
                      CleanDialogActionButtons(
                        actionTitle: 'Try again',
                        textColor: const Color(0XFF27ae61),
                        onPressed: () {},
                      ),
                    ],
                  ),
                );
              },
              child: const Center(child: Text('Second Example')),
            ),
          ],
        ),
      ),
    );
  }
}
